boa/src/builtins/set/set_iterator.rs

```diff
@@ -0,0 +1,149 @@
+use crate::{
+    builtins::function::make_builtin_fn,
+    builtins::iterable::create_iter_result_object,
+    builtins::Array,
+    builtins::Value,
+    object::{GcObject, ObjectData},
+    property::{Attribute, DataDescriptor},
+    BoaProfiler, Context, Result,
+};
+use gc::{Finalize, Trace};
+
+#[derive(Debug, Clone, Finalize, Trace)]
+pub enum SetIterationKind {
+    Value,
+    KeyAndValue,
+}
+
+/// The Set Iterator object represents an iteration over a set. It implements the iterator protocol.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-set-iterator-objects
+#[derive(Debug, Clone, Finalize, Trace)]
+pub struct SetIterator {
+    iterated_set: Value,
+    set_next_index: usize,
+    set_iteration_kind: SetIterationKind,
+}
+
+impl SetIterator {
+    pub(crate) const NAME: &'static str = "SetIterator";
+
+    fn new(set: Value, kind: SetIterationKind) -> Self {
+        SetIterator {
+            iterated_set: set,
+            set_next_index: 0,
+            set_iteration_kind: kind,
+        }
+    }
+
+    /// Abstract operation CreateSetIterator( set, kind )
+    ///
+    /// Creates a new iterator over the given set.
+    ///
+    /// More information:
+    ///  - [ECMA reference][spec]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/11.0/index.html#sec-createsetiterator
+    pub(crate) fn create_set_iterator(
+        context: &Context,
+        set: Value,
+        kind: SetIterationKind,
+    ) -> Result<Value> {
+        let set_iterator = Value::new_object(context);
+        set_iterator.set_data(ObjectData::SetIterator(Self::new(set, kind)));
+        set_iterator
+            .as_object()
+            .expect("set iterator object")
+            .set_prototype_instance(context.iterator_prototypes().set_iterator().into());
+        Ok(set_iterator)
```

Comment:
```suggestion
    ) -> Value {
        let set_iterator = Value::new_object(context);
        set_iterator.set_data(ObjectData::SetIterator(Self::new(set, kind)));
        set_iterator
            .as_object()
            .expect("set iterator object")
            .set_prototype_instance(context.iterator_prototypes().set_iterator().into());
        set_iterator
```
Since we cant throw I think its best to remove the unnecessary Result wrap.